sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -34,6 +34,7 @@
 from sklearn.utils.fast_dict import IntFloatDict
 from sklearn.utils.testing import assert_array_equal
 from sklearn.utils.testing import assert_warns
+from sklearn.utils.tests.test_validation import DummyMemory, WrongDummyMemory
```

Comment:
I would need @jnothman to tell me if doing this is fine.
I am not sure anymore that it is usually done in that way (even if we avoid code duplication)